include/RAJA/policy/cuda/reduce.hpp

```diff
@@ -70,21 +72,99 @@
 
 #include <cuda.h>
 
+#if defined(RAJA_ENABLE_OPENMP)
+#include <omp.h>
+#endif
+
 namespace RAJA
 {
 
-// HIDDEN namespace to encapsulate helper functions
-namespace HIDDEN
+namespace reduce
+{
+
+template <typename T>
+struct atomic_operators
+{
+
+};
+
+template <typename T>
+struct atomic_operators<sum<T>>
+{
+  RAJA_HOST_DEVICE RAJA_INLINE
+  void operator()(T &val, const T v)
+  {
+#ifdef __CUDA_ARCH__
+    RAJA::_atomicAdd(&val, v);
+#else
+    val += v;
```

Comment:
Under RAJA/pattern/reduce.hpp we do have sum, min, max, minloc, and maxloc reducer functions already defined.

I'd recommend using them instead of hard-coding reduction operations